doc/query.md

```diff
@@ -39,27 +39,60 @@ Secondly, these documents are in the datastore:
 ```java
 { "name": "mike", 
   "age": 12, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Mike goes to middle school and likes reading books." };
 
 { "name": "mike", 
   "age": 34, 
-  "pet": {"species": "dog"} };
+  "pet": {"species": "dog"},
+  "comment": "Mike is a doctor and likes reading books." };
 
 { "name": "fred", 
   "age": 23, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Fred works for a startup out of his home office." };
 ```
 
-### Creating indexes
+### Creating Indexes
 
 In order to query documents, indexes need to be created over
-the fields to be queried against.
+the fields to be queried against.  Currently we support two types of indexes.  The first, a JSON index, is used by query clauses containing comparison operators like `$eq`, `$lt`, `$gt`, etc.  Query clauses containing these operators rely on standard SQL syntax along with JSON indexes to provide query results.  The second, a TEXT index, uses SQLite's full text search (FTS) engine.  A query clause containing a `$text` operator with a `$search` operator uses [SQLite FTS SQL syntax] [ftsHome] along with a TEXT index to provide query results.
 
-Use `ensureIndexed(List<Object> fieldNames, String indexName)` to create indexes. These indexes are persistent across application restarts as they are saved to disk. They are kept up to date documents change; there's no need to call `ensureIndexed(List<Object> fieldNames, String indexName)` each time your applications starts, though there is no harm in doing so.
+Using the datastore documents example above, a query like `{ "name" : { "$eq" : "mike" } }` would benefit from a JSON index on the `name` field.  Whereas a query like `{ "$text" : { "$search" : "doctor books" } }` would _require_ a TEXT index on the `comment` field but could include as many fields as desired.  In other words, A TEXT index can be used to search for terms, phrases, prefixes with wildcards, etc.  Query syntax details are covered later in this document.
```

Comment:
Simplified as suggested.  What I meant by "could include as many fields as desired" was just that you don't have to have an index just on the comment field.  I see now that that is unnecessary here so I removed it.  Done in ee198020c5b7e28d8fcf044f68f77f22fa1db7e2.